src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs

```diff
@@ -100,23 +106,31 @@ impl VisualizationChooser {
 
             // === Input Processing ===
 
-            eval frp.input.set_entries ([model,menu](alternatives) {
-                model.visualization_alternatives.replace(alternatives.clone());
-                let alternatives:list_view::entry::AnyModelProvider  = alternatives.clone().into();
-                menu.set_entries.emit(alternatives);
-            });
+            eval  frp.set_icon_size ((size) menu.set_icon_size.emit(size) );
+            eval  frp.set_icon_padding ((size) menu.set_icon_padding.emit(size) );
+            eval_ frp.hide_selection_menu ( menu.hide_selection_menu.emit(()) );
+            eval  frp.set_menu_offset_y ((offset) menu.set_menu_offset_y.emit(offset) );
+
+            _set_selected <- frp.input.set_selected.map2(&frp.output.entries,
+                f!([menu](selected,entries) {
+                    if let Some(selected) = selected {
+                        let ix = entries.iter().position(|item| item == selected);
+                        menu.set_selected.emit(ix);
+                    }
+                })
+            );
+
+
+            // === Showing Entries ===
+
 
-            eval frp.set_icon_size ((size) menu.set_icon_size.emit(size) );
-            eval frp.set_icon_padding ((size) menu.set_icon_padding.emit(size) );
-            eval frp.hide_selection_menu ((size) menu.hide_selection_menu.emit(size) );
-            eval frp.set_menu_offset_y ((offset) menu.set_menu_offset_y.emit(offset) );
+            frp.source.entries <+ menu.menu_visible.gate(&menu.menu_visible).map(f_!([model] {
+                let entries                                     = Rc::new(model.entries());
+                let provider:list_view::entry::AnyModelProvider = entries.clone_ref().into();
```

Comment:
much nicer layout of code would be if you'd write it as `let provider = list_view::entry::AnyModelProvider::from(entries.clone_ref());`